Fix and accelerate clojure.core completion

mainej opened the pull request:
Make completion of clojure.core more consistent by using queries. Returns slightly fewer results, because of de-duplication. Also, change filtering order to slightly improve performance of completion, especially for completions with more given characters.

This is an extension of #764.

ericdallo commented on lib/src/clojure_lsp/queries.clj:
```diff
@@ -412,32 +412,28 @@
     (catch Throwable e
       (log/error e "can't find references"))))
 
+(defn xf-var-defs [include-private?]
```
```suggestion
(defn ^:private xf-var-defs [include-private?]
```

mainej replied:
It's used by another ns.

ericdallo replied:
yeah, I just saw it

ericdallo commented:
Code looks good!
Would you mind adding integration tests for the clojure.core/cljs.core completion, the integration tests are simple ATM?

mainej commented:
> Would you mind adding integration tests
Yep, happy to.

mainej commented:
@ericdallo I added integration tests. They dump a TON of data into the test output. The entire response from the server is logged. I can't think of a way to suppress that logging. Do you still want them? IMO, the unit tests cover the same thing. 

On that subject, I noticed the unit tests were missing a test for cljs.core completion, so I added that too.

ericdallo commented:
I see... Well, I'm ok adding only unit for this since these changes affect only completion data and not something a integration test would catch but a unit not.

ericdallo reviewed (approved):
Thank you!
